### PR TITLE
[WIP] Fix xref

### PR DESCRIPTION
**Pull request by sebastiencs**
\+ Change `xref-backend-definitions`, `lsp--get-references` and `lsp--get-definitions`

Following #196

### CODE REVIEW

**Comment by sebastiencs** [comment]
I will refactor the code and fix apropos

**Comment by sebastiencs** [comment]
It should be ok now

File: lsp-methods.el [hunk 1]
```diff
@@ -1113,30 +1113,70 @@ https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#co
...
+                            (max (min end len) 0)
+                            'highlight t line)
+    (xref-make (or line filename)
+               (xref-make-file-location filename
+                                        (1+ (gethash "line" pos-start))
+                                        (gethash "character" pos-start)))))
+
+(defun lsp--get-xrefs-in-file (file)
+  "Return all references that contain a file.
+FILE is a cons where its car is the filename and the cdr is a list of locations.
```
Comment (vibhavp) [inline lsp-methods.el]: Why is the CDR a list of locations? If we're only removing the "file://" prefix from the filename string, shouldn't that be just a single element? Eg: `(("file:///etc/foo" "/etc/foo") ("file:///etc/bar" "/etc/bar"))`

Comment (sebastiencs) [reply lsp-methods.el]: The param `file` is an element returned by `seq-group-by`:
```elisp
(seq-group-by (lambda (n) (if (evenp n) "even" "odd")) '(1 2 3 4 5 6))
=> (("odd" 1 3 5) ("even" 2 4 6))
```
The car of each element is the result of the function given to `seq-group-by` and their cdr are unchanged element from the param `SEQUENCE`.
In our case elements are locations:
```elisp
("/path/file" ("file:///path/file" range) ("file:///path/file" another_range))
```

Comment (vibhavp) [reply lsp-methods.el]: Ah, got it. Thanks

**Pull request merged**